File: Src/Icm.Core/AI Search/INode.vb

Namespace Icm.Collections.Generic

    Public Interface INode
        Property Value As String
        Property CameFrom As INode
        Property g As Integer
        Function h() As Integer

        Property Neighbors As IEnumerable(Of INode)
        Function IsGoal() As Boolean
        Function Distance(ByVal other As INode) As Integer
    End Interface
End Namespace

File: Src/Icm.Core/AI Search/INode.cs
using System.Collections.Generic;

namespace Icm.Collections.Generic
{

	public interface INode
	{
		string Value { get; set; }
		INode CameFrom { get; set; }
		int g { get; set; }
		int h();

		IEnumerable<INode> Neighbors { get; set; }
		bool IsGoal();
		int Distance(INode other);
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
